sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -757,4 +767,80 @@
             return monoError(logger, ex);
         }
     }
+
+    private static final Duration DEFAULT_POLL_INTERVAL = Duration.ofSeconds(5);
+
+    /**
+     * Long Running Operation for the purchaseSearch.
+     *
+     * @param searchId ID of the search
+     * @param lroDuration Timelapse of the poll request.
+     * @return A {@link Mono} containing a {@link Response} for the operation
+     */
+
+    public PollerFlux<Void, PhoneNumberSearch> beginPurchaseSearch(String searchId, Duration lroDuration) {
```

Comment:
Also, public methods should a @ServiceClient decorator, like this https://github.com/Azure/azure-sdk-for-java/blob/faae145a6b80f3dae615e8e4f46ba6bd50681f32/sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java#L778